app/src/main/javascript/bundles/experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -42,28 +37,24 @@ class TSnePlotViewRoute extends React.Component {
       selectedColourBy: cellTypeValue ? cellTypeValue.toLowerCase() : this.props.ks[Math.round((this.props.ks.length -1) / 2)].toString(),
       highlightClusters: [],
       experimentAccession: this.props.experimentAccession,
-      selectedColourByCategory: isNaN(search.colourBy) ? `metadata` :  `clusters`,
+      selectedColourByCategory: isNaN(search.colourBy) ? METADATA_PLOT :  CLUSTERS_PLOT,
       selectedClusterId: cellTypeValue ? cellTypeValue.toLowerCase() : this.props.ks[Math.round((this.props.ks.length -1) / 2)].toString(),
       selectedClusterIdOption: ``
     }
   }
 
   render() {
     const { location, match, history } = this.props
-    const { atlasUrl, suggesterEndpoint, initialCellTypeValues, defaultPlotMethodAndParameterisation } = this.props
+    const { atlasUrl, suggesterEndpoint, defaultPlotMethodAndParameterisation } = this.props
     const { species, experimentAccession, accessKey, ks, ksWithMarkerGenes, plotTypesAndOptions, metadata, anatomogram } = this.props
     const search = URI(location.search).search(true)
 
-    const plotTypeDropdown =  [
-      {
-        plotType: Object.keys(defaultPlotMethodAndParameterisation)[0],
-        plotOptions: plotTypesAndOptions[Object.keys(defaultPlotMethodAndParameterisation)[0]]
-      },
-      {
-        plotType: Object.keys(defaultPlotMethodAndParameterisation)[1],
-        plotOptions: plotTypesAndOptions[Object.keys(defaultPlotMethodAndParameterisation)[1]]
-      }
-    ]
+    let plotTypeDropdown =  []
+    Object.keys(defaultPlotMethodAndParameterisation).map(plot => plotTypeDropdown.push(
+    {
+      plotType: plot,
+          plotOptions: plotTypesAndOptions[plot]
+    }))
```

Comment:
Eliminates the creation an empty object first and use of the `push` method later. `map` is already doing all of that.
```suggestion
let plotTypeDropdown =
        Object.keys(defaultPlotMethodAndParameterisation)
            .map(plot => ({
              plotType: plot,
              plotOptions: plotTypesAndOptions[plot]
            }));
```